accepted/2021/statics-in-interfaces/README.md

```diff
@@ -8,6 +8,793 @@ C# is looking at enabling static abstract members in interfaces (https://github.
 
 ## Designs
 
+The overall design has been influenced by the existing .NET Surface Area, the Swift numeric protocols (https://developer.apple.com/documentation/swift/swift_standard_library/numbers_and_basic_values/numeric_protocols), and the Rust op traits (https://doc.rust-lang.org/std/ops/index.html). It is broken up into a few layers describing a hierarchy of interfaces and what types implement them. The interfaces aim to be reusable by a variety of types and so do not provide guarantees around concepts like associativity, commutativity or distributivity. Instead, these concepts may become available to tooling in the future via another mechanism, such as attributes on the respective operator implementations.
+
+### Base Interfaces
+
+The base interfaces each describe one or more core operations that are used to build up the core interfaces listed later. In many cases, such as for addition, subtraction, multiplication, and division, there is only a single operator per interface. This split is important in order for them to be reusable against downstream types, like `DateTime` where addition does not imply subtraction or `Matrix` where multiplication does not imply division. These interfaces are not expected to be used by many users and instead primarily by developers looking at defining their own abstractions, as such we are not concerned with adding simplifying interfaces like `IAddable<TSelf> : IAddable<TSelf, TSelf>`. Likewise, there is not a concern with exposing `IAddable<TSelf, TOther, TResult>` and instead we will require `TSelf` and `TResult` be the same. This requirement greatly simplifies implementation and consumption of the types.
+
+```csharp
```

Comment:
Could add this?

```markdown
<!--
@startuml
IParseable <|-- ISpanParseable
IEquatable <|-- IEquatableOperators

IComparable <|-- IComparableOperators
IEquatableOperators <|-- IComparableOperators

IAddable          <|-- INumber
IDecrementable    <|-- INumber
IDivisable        <|-- INumber
IIncrementable    <|-- INumber
IMultipliable     <|-- INumber
IRemainder        <|-- INumber
ISubtractable     <|-- INumber

INumber    <|-- ISignedNumber
INegatable <|-- ISignedNumber

INumber <|-- IUnsignedNumber
INumber <|-- IBinaryNumber

IBinaryNumber <|-- ISignedBinaryNumber
ISignedNumber <|-- ISignedBinaryNumber

IBinaryNumber   <|-- IUnsignedBinaryNumber
IUnsignedNumber <|-- IUnsignedBinaryNumber

IBinaryNumber <|-- IBinaryInteger

IBinaryInteger      <|-- ISignedBinaryInteger
ISignedBinaryNumber <|-- ISignedBinaryInteger

IBinaryInteger        <|-- IUnsignedBinaryInteger
IUnsignedBinaryNumber <|-- IUnsignedBinaryInteger

ISignedNumber <|-- IFloatingPoint

IFloatingPoint <|-- IBinaryFloatingPoint
IBinaryNumber  <|-- IBinaryFloatingPoint

IFloatingPoint <|-- IDecimalFloatingPoint

interface IParseable<TSelf>
interface ISpanParseable<TSelf>
interface IEquatableOperators<TSelf, TOther>
interface IComparableOperators<TSelf, TOther>
interface IAddable<TSelf, TOther>
interface ISubtractable<TSelf, TOther>
interface IIncrementable<TSelf>
interface IDecrementable<TSelf>
interface IMultipliable<TSelf, TOther>
interface IDivisable<TSelf, TOther>
interface IRemainder<TSelf, TOther>
interface INegatable<TSelf>
interface INumber<TSelf>
interface ISignedNumber<TSelf>
interface IUnsignedNumber<TSelf>
interface IBinaryNumber<TSelf>
interface ISignedBinaryNumber<TSelf>
interface IUnsignedBinaryNumber<TSelf>
interface IBinaryInteger<TSelf>
interface ISignedBinaryInteger<TSelf>
interface IUnsignedBinaryInteger<TSelf>
interface IFloatingPoint<TSelf>
interface IBinaryFloatingPoint<TSelf>
interface IDecimalFloatingPoint<TSelf>
@enduml
-->

![UML](http://www.plantuml.com/plantuml/svg/VPHHJiCm44J_VOeSWDwWGWKAaZzeAr8Eu2RROCbnWkCW8N5ufAOCNhHilshFs_7QoivcoYxbkG_67boQo9q2LUklrQgozUWYzenzV9zTBcIizoCbbuSq6MCVXdvqgQ2mKM12gK8RUzUs5qVy5dWtzoTAncwfITHJl3WAil_mKrdFPHlLwkSvP3y63mCklr3lV6mfoURrVCh9DLckDjS_lqBjkqZjJ_MEEhmuKrdneht6gLIvUE-ZIvyeB9VCcd7iE8PfTXNlXxFO5LfNRTazRCpK0S4Qxy_x12uqBz2gkNWvb8WNKIl4DtuAWyi-TeV1nsmCNxE7u2H_C07KBRVK-Dw5FwYFcTBPDLGX7cXzh2cSRqiDKI7hmi0lq4rrtEStIWmMGa2Xh-6W41X25UD18DnXIxgEfD2FGHmf3595OH0EGYlBDo08-Dna6UCf0mWItHwGVeYEyV7JpmAb7wPmV2B-xKQdn1a2jg7Opdtu1W00)
```

This will render a UML diagram:

![UML](http://www.plantuml.com/plantuml/svg/VPHHJiCm44J_VOeSWDwWGWKAaZzeAr8Eu2RROCbnWkCW8N5ufAOCNhHilshFs_7QoivcoYxbkG_67boQo9q2LUklrQgozUWYzenzV9zTBcIizoCbbuSq6MCVXdvqgQ2mKM12gK8RUzUs5qVy5dWtzoTAncwfITHJl3WAil_mKrdFPHlLwkSvP3y63mCklr3lV6mfoURrVCh9DLckDjS_lqBjkqZjJ_MEEhmuKrdneht6gLIvUE-ZIvyeB9VCcd7iE8PfTXNlXxFO5LfNRTazRCpK0S4Qxy_x12uqBz2gkNWvb8WNKIl4DtuAWyi-TeV1nsmCNxE7u2H_C07KBRVK-Dw5FwYFcTBPDLGX7cXzh2cSRqiDKI7hmi0lq4rrtEStIWmMGa2Xh-6W41X25UD18DnXIxgEfD2FGHmf3595OH0EGYlBDo08-Dna6UCf0mWItHwGVeYEyV7JpmAb7wPmV2B-xKQdn1a2jg7Opdtu1W00)

Reply:
This is great, I didn't know it was supported.

Reply:
It's not really; but you'll see once you inspect the snippet closely :-)

Reply:
Tricky tricky